src/llmq/quorums_instantsend.cpp

```diff
@@ -887,8 +941,23 @@ void CInstantSendManager::HandleFullyConfirmedBlock(const CBlockIndex* pindex)
             }
         }
 
-        // Retry all not yet locked mempool TXs and TX which where mined after the fully confirmed block
-        pendingRetryAllTxs = true;
+        // Find all previously unlocked TXs that got locked by this fully confirmed (ChainLock) block and remove them
+        // from the nonLockedTxs map. Also collect all children of these TXs and mark them for retrying of IS locking.
+        std::vector<uint256> toRemove;
+        for (auto& p : nonLockedTxs) {
+            auto& info = p.second;
+            if (!p.second.pindexMined) {
+                continue;
+            }
+
+            if (pindex->GetAncestor(info.pindexMined->nHeight) == info.pindexMined) {
```

Comment:
nit: `info` isn't really used here it seems, could compact this down to smth like
```
auto& pindexMined = p.second.pindexMined;
if (pindexMined && pindex->GetAncestor(pindexMined->nHeight) == pindexMined) {
```
and drop `info` and `if` above